### DISCUSSION TITLE
API / Exist.io

### DISCUSSION THREAD

**Discussion by RedApCi** [iSoron/uhabits#838] [Feature Requests] [answered]
Hi, I use a service (exist.io) that aggregates personal data (activity tracker/to-do list/ calendar etc) to find useful correlations. Integration with loop could be fantastic for this, but they say it is not currently possible because Loop has no API:
https://changemap.co/hellocode/exist/task/3791-loop-habit-tracker/
I saw a post of yours on Reddit from a while ago saying you would consider an API, is this now available or planned for the future?
Really enjoying Loop, the best habit tracker I have found, Thanks!

**Comment by iSoron** [comment, marked as answer]
I agree with their assessment that such an integration is not possible. This is by design. As of Loop 1.8, we never send user data to any centralized server, so it is impossible for us to provide it to anyone else (via an API or any other means); we just don't have the data. When device sync becomes available, user data will be end-to-end encrypted, with the encryption keys remaining only in your phone, so it would still be impossible for us to share it.

**Comment by 4unkur** [comment]
Hello @iSoron

How can I automate data export somehow? Any ideas? I just want to send the Exported data to my server everyday. This is for personal use only
